src/test/java/seedu/address/model/person/GroupTest.java

```diff
@@ -0,0 +1,39 @@
+package seedu.address.model.person;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+public class GroupTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Group(null));
+    }
+
+    @Test
+    public void constructor_invalidName_throwsIllegalArgumentException() {
+        String invalidName = "";
+        assertThrows(IllegalArgumentException.class, () -> new Group(invalidName));
+    }
+
+    @Test
+    public void isValidGroup() {
+        // null name
+        assertThrows(NullPointerException.class, () -> Group.isValidGroup(null));
+
+        // invalid name
+        assertFalse(Group.isValidGroup("")); // empty string
+        assertFalse(Group.isValidGroup(" ")); // spaces only
```

Comment:
can try starting with a/multiple space(s) as well for example " CS3243" and "   CS3243"